gix-object v0.44.0 fail to compile

### Current behavior 😯

https://github.com/cargo-bins/simple-git/actions/runs/10517678165/job/29142235436?pr=35

```
error[E0599]: `(&[u8], impl Parser<_, <_ as Stream>::Slice, _>, &[u8], fn(&mut _) -> Result<..., ...> {rest::<..., ...>})` is not an iterator
  --> /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gix-object-0.44.0/src/tag/decode.rs:59:26
   |
53 | /                     (
54 | |                         &PGP_SIGNATURE_BEGIN[1..],
55 | |                         take_until(0.., PGP_SIGNATURE_END),
56 | |                         PGP_SIGNATURE_END,
57 | |                         rest,
58 | |                     )
59 | |                         .take()
   | |                         -^^^^ `(&[u8], impl Parser<_, <_ as Stream>::Slice, _>, &[u8], fn(&mut _) -> Result<..., ...> {rest::<..., ...>})` is not an iterator
   | |_________________________|
   |
   |
   = note: the full type name has been written to '/Users/runner/work/simple-git/simple-git/target/aarch64-apple-darwin/debug/deps/gix_object-976e30f84e72c296.long-type-4190708012872330450.txt'
   = note: consider using `--verbose` to print the full type name to the console
   = note: the full name for the type has been written to '/Users/runner/work/simple-git/simple-git/target/aarch64-apple-darwin/debug/deps/gix_object-976e30f84e72c296.long-type-4190708012872330450.txt'
   = note: consider using `--verbose` to print the full type name to the console
   = note: the following trait bounds were not satisfied:
           `(&[u8], impl winnow::Parser<_, <_ as Stream>::Slice, _>, &'static [u8], for<'a> fn(&'a mut _) -> Result<<_ as Stream>::Slice, ErrMode<_>> {rest::<_, _>}): Iterator`
           which is required by `&mut (&[u8], impl winnow::Parser<_, <_ as Stream>::Slice, _>, &'static [u8], for<'a> fn(&'a mut _) -> Result<<_ as Stream>::Slice, ErrMode<_>> {rest::<_, _>}): Iterator`

error[E0599]: the method `take` exists for tuple `(impl Parser<_, <_ as Stream>::Slice, _>, &[u8], Map<..., ..., ..., ..., ..., ...>)`, but its trait bounds were not satisfied
  --> /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gix-object-0.44.0/src/parse.rs:25:14
   |
20 | /         (
21 | |             take_till(1.., NL),
22 | |             NL,
23 | |             repeat(1.., terminated((SPACE, take_until(0.., NL)), NL)).map(|()| ()),
24 | |         )
25 | |             .take()
   | |             -^^^^ method cannot be called due to unsatisfied trait bounds
   | |_____________|
   |
   |
   = note: the full type name has been written to '/Users/runner/work/simple-git/simple-git/target/aarch64-apple-darwin/debug/deps/gix_object-976e30f84e72c296.long-type-16570815735586947285.txt'
   = note: consider using `--verbose` to print the full type name to the console
   = note: the following trait bounds were not satisfied:
           `(impl winnow::Parser<_, <_ as Stream>::Slice, _>, &'static [u8], winnow::combinator::Map<winnow::combinator::Repeat<impl winnow::Parser<_, (_, _), _>, _, (_, _), (), _>, {closure@/Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gix-object-0.44.0/src/parse.rs:23:75: 23:79}, _, (), (), _>): Iterator`
           which is required by `&mut (impl winnow::Parser<_, <_ as Stream>::Slice, _>, &'static [u8], winnow::combinator::Map<winnow::combinator::Repeat<impl winnow::Parser<_, (_, _), _>, _, (_, _), (), _>, {closure@/Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gix-object-0.44.0/src/parse.rs:23:75: 23:79}, _, (), (), _>): Iterator`
```

### Expected behavior 🤔

Should compile without error

### Git behavior

_No response_

### Steps to reproduce 🕹

_No response_

--- a/gix-object/Cargo.toml
+++ b/gix-object/Cargo.toml
@@ -50,7 +50,7 @@ bstr = { version = "1.3.0", default-features = false, features = [
     "std",
     "unicode",
 ] }
-winnow = { version = "0.6", features = ["simd"] }
+winnow = { version = "0.6.18", features = ["simd"] }
 smallvec = { version = "1.4.0", features = ["write"] }
 serde = { version = "1.0.114", optional = true, default-features = false, features = [
     "derive",

